src/main/java/org/kohsuke/github/Requester.java

```diff
@@ -758,8 +758,11 @@ private InputStream wrapStream(InputStream in) throws IOException {
                 IOUtils.closeQuietly(es);
             }
         }
-        if (responseCode == HttpURLConnection.HTTP_UNAUTHORIZED) // 401 / Unauthorized == bad creds
-            throw e;
+        if (responseCode == HttpURLConnection.HTTP_UNAUTHORIZED) // 401 Unauthorized == bad creds
+            if(uc.getHeaderField("X-GitHub-OTP") != null)
```

Comment:
Add some comments about why this code path exists.

Reply:
done